File: runtime/flutter/rendering/diagnosticsDebugCreator.ts
import { IList } from "../../dart/core/list";
import { Type } from "../../dart/core/type";
import { DiagnosticLevel } from "../foundation/diagnosticLevel";
import { IDiagnosticsNode } from "../foundation/diagnosticsNode";
import { IDiagnosticsProperty } from "../foundation/diagnosticsProperty";
import { IDiagnosticsSerializationDelegate } from "../foundation/diagnosticsSerializationDelegate";
import { DiagnosticsTreeStyle } from "../foundation/diagnosticsTreeStyle";
import { ITextTreeConfiguration } from "../foundation/textTreeConfiguration";
declare const flutter: {
    rendering: {
        diagnosticsDebugCreator: (
            this: void,
            diagnosticsDebugCreator: IDiagnosticsDebugCreator,
            value: Object
        ) => IDiagnosticsDebugCreator;
    };
};
export interface IDiagnosticsDebugCreator {
    expandableValue: boolean;
    allowWrap: boolean;
    allowNameWrap: boolean;
    ifNull: string | undefined;
    ifEmpty: string | undefined;
    tooltip: string | undefined;
    missingIfNull: boolean;
    defaultValue: Object | undefined;
    name: string | undefined;
    showSeparator: boolean;
    showName: boolean;
    linePrefix: string | undefined;
    style: DiagnosticsTreeStyle | undefined;
    toJsonMap: (delegate: IDiagnosticsSerializationDelegate) => {
        [index: string]: Object | undefined;
    };
    valueToString: (props?: {
        parentConfiguration?: ITextTreeConfiguration | undefined;
    }) => string;
    toDescription: (props?: {
        parentConfiguration?: ITextTreeConfiguration | undefined;
    }) => string;
    getProperties: () => IList<IDiagnosticsNode>;
    getChildren: () => IList<IDiagnosticsNode>;
    getPropertyType: () => Type;
    getValue: () => Object | undefined;
    getException: () => Object | undefined;
    getLevel: () => DiagnosticLevel;
    isFiltered: (minLevel: DiagnosticLevel) => boolean;
    toString: (props: {
        minLevel: DiagnosticLevel;
        parentConfiguration?: ITextTreeConfiguration | undefined;
    }) => string;
    toStringDeep: (props: {
        minLevel: DiagnosticLevel;
        parentConfiguration?: ITextTreeConfiguration | undefined;
        prefixLineOne: string;
        prefixOtherLines?: string | undefined;
    }) => string;
    getEmptyBodyDescription: () => string | undefined;
    getAllowWrap: () => boolean;
    getAllowNameWrap: () => boolean;
    getAllowTruncate: () => boolean;
    getTextTreeConfiguration: () => ITextTreeConfiguration | undefined;
    getHashCode: () => number;
}
export class DiagnosticsDebugCreator implements IDiagnosticsProperty<Object> {
    public readonly expandableValue: boolean = undefined as any;
    public readonly allowWrap: boolean = undefined as any;
    public readonly allowNameWrap: boolean = undefined as any;
    public readonly ifNull: string | undefined = undefined as any;
    public readonly ifEmpty: string | undefined = undefined as any;
    public readonly tooltip: string | undefined = undefined as any;
    public readonly missingIfNull: boolean = undefined as any;
    public readonly defaultValue: Object | undefined = undefined as any;
    public readonly name: string | undefined = undefined as any;
    public readonly showSeparator: boolean = undefined as any;
    public readonly showName: boolean = undefined as any;
    public readonly linePrefix: string | undefined = undefined as any;
    public readonly style: DiagnosticsTreeStyle | undefined = undefined as any;
    public constructor(value: Object) {
        flutter.rendering.diagnosticsDebugCreator(this, value);
    }
    private readonly _dart_toJsonMap: (
        delegate: IDiagnosticsSerializationDelegate
    ) => { [index: string]: Object | undefined } = undefined as any;
    private readonly _dart_valueToString: (props?: {
        parentConfiguration?: ITextTreeConfiguration | undefined;
    }) => string = undefined as any;
    private readonly _dart_toDescription: (props?: {
        parentConfiguration?: ITextTreeConfiguration | undefined;
    }) => string = undefined as any;
    private readonly _dart_getProperties: () => IList<IDiagnosticsNode> =
        undefined as any;
    private readonly _dart_getChildren: () => IList<IDiagnosticsNode> =
        undefined as any;
    private readonly _dart_getPropertyType: () => Type = undefined as any;
    private readonly _dart_getValue: () => Object | undefined =
        undefined as any;
    private readonly _dart_getException: () => Object | undefined =
        undefined as any;
    private readonly _dart_getLevel: () => DiagnosticLevel = undefined as any;
    private readonly _dart_isFiltered: (minLevel: DiagnosticLevel) => boolean =
        undefined as any;
    private readonly _dart_toString: (props: {
        minLevel: DiagnosticLevel;
        parentConfiguration?: ITextTreeConfiguration | undefined;
    }) => string = undefined as any;
    private readonly _dart_toStringDeep: (props: {
        minLevel: DiagnosticLevel;
        parentConfiguration?: ITextTreeConfiguration | undefined;
        prefixLineOne: string;
        prefixOtherLines?: string | undefined;
    }) => string = undefined as any;
    private readonly _dart_getEmptyBodyDescription: () => string | undefined =
        undefined as any;
    private readonly _dart_getAllowWrap: () => boolean = undefined as any;
    private readonly _dart_getAllowNameWrap: () => boolean = undefined as any;
    private readonly _dart_getAllowTruncate: () => boolean = undefined as any;
    private readonly _dart_getTextTreeConfiguration: () =>
        | ITextTreeConfiguration
        | undefined = undefined as any;
    private readonly _dart_getHashCode: () => number = undefined as any;
    public toJsonMap(delegate: IDiagnosticsSerializationDelegate): {
        [index: string]: Object | undefined;
    } {
        return this._dart_toJsonMap(delegate);
    }
    public valueToString(props?: {
        parentConfiguration?: ITextTreeConfiguration | undefined;
    }): string {
        return this._dart_valueToString(props);
    }
    public toDescription(props?: {
        parentConfiguration?: ITextTreeConfiguration | undefined;
    }): string {
        return this._dart_toDescription(props);
    }
    public getProperties(): IList<IDiagnosticsNode> {
        return this._dart_getProperties();
    }
    public getChildren(): IList<IDiagnosticsNode> {
        return this._dart_getChildren();
    }
    public getPropertyType(): Type {
        return this._dart_getPropertyType();
    }
    public getValue(): Object | undefined {
        return this._dart_getValue();
    }
    public getException(): Object | undefined {
        return this._dart_getException();
    }
    public getLevel(): DiagnosticLevel {
        return this._dart_getLevel();
    }
    public isFiltered(minLevel: DiagnosticLevel): boolean {
        return this._dart_isFiltered(minLevel);
    }
    public toString(props: {
        minLevel?: DiagnosticLevel;
        parentConfiguration?: ITextTreeConfiguration | undefined;
    }): string {
        return this._dart_toString({
            ...toStringDefaultProps,
            ...props,
        });
    }
    public toStringDeep(props: {
        minLevel?: DiagnosticLevel;
        parentConfiguration?: ITextTreeConfiguration | undefined;
        prefixLineOne?: string;
        prefixOtherLines?: string | undefined;
    }): string {
        return this._dart_toStringDeep({
            ...toStringDeepDefaultProps,
            ...props,
        });
    }
    public getEmptyBodyDescription(): string | undefined {
        return this._dart_getEmptyBodyDescription();
    }
    public getAllowWrap(): boolean {
        return this._dart_getAllowWrap();
    }
    public getAllowNameWrap(): boolean {
        return this._dart_getAllowNameWrap();
    }
    public getAllowTruncate(): boolean {
        return this._dart_getAllowTruncate();
    }
    public getTextTreeConfiguration(): ITextTreeConfiguration | undefined {
        return this._dart_getTextTreeConfiguration();
    }
    public getHashCode(): number {
        return this._dart_getHashCode();
    }
}
const toStringDefaultProps = {
    minLevel: DiagnosticLevel.info,
};
const toStringDeepDefaultProps = {
    minLevel: DiagnosticLevel.debug,
    prefixLineOne: "",
};

File: lib/cfr/builtins/libs/flutter/rendering/diagnosticsDebugCreator.dart
import 'dart:core';

import 'package:flutter/src/foundation/diagnostics.dart';
import 'package:flutter/src/rendering/object.dart';

import 'package:hydro_sdk/cfr/runtimeSupport.dart';

class VMManagedDiagnosticsDebugCreator
    extends VMManagedBox<DiagnosticsDebugCreator> {
  VMManagedDiagnosticsDebugCreator(
      {required this.table, required this.vmObject, required this.hydroState})
      : super(
          table: table,
          vmObject: vmObject,
          hydroState: hydroState,
        ) {
    table['expandableValue'] = vmObject.expandableValue;
    table['allowWrap'] = vmObject.allowWrap;
    table['allowNameWrap'] = vmObject.allowNameWrap;
    table['ifNull'] = vmObject.ifNull;
    table['ifEmpty'] = vmObject.ifEmpty;
    table['tooltip'] = vmObject.tooltip;
    table['missingIfNull'] = vmObject.missingIfNull;
    table['defaultValue'] = maybeBoxObject<Object?>(
        object: vmObject.defaultValue,
        hydroState: hydroState,
        table: HydroTable());
    table['name'] = vmObject.name;
    table['showSeparator'] = vmObject.showSeparator;
    table['showName'] = vmObject.showName;
    table['linePrefix'] = vmObject.linePrefix;
    table['style'] = DiagnosticsTreeStyle.values.indexWhere((x) {
      return x == vmObject.style;
    });
    table['toJsonMap'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        maybeBoxObject<Map>(
            object: vmObject.toJsonMap(maybeUnBoxAndBuildArgument<
                DiagnosticsSerializationDelegate,
                dynamic>(luaCallerArguments[1], parentState: hydroState)),
            hydroState: hydroState,
            table: HydroTable()),
      ];
    });
    table['valueToString'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        vmObject.valueToString(
            parentConfiguration:
                maybeUnBoxAndBuildArgument<TextTreeConfiguration?, dynamic>(
                    luaCallerArguments.length >= 2
                        ? luaCallerArguments[1]['parentConfiguration']
                        : null,
                    parentState: hydroState)),
      ];
    });
    table['toDescription'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        vmObject.toDescription(
            parentConfiguration:
                maybeUnBoxAndBuildArgument<TextTreeConfiguration?, dynamic>(
                    luaCallerArguments.length >= 2
                        ? luaCallerArguments[1]['parentConfiguration']
                        : null,
                    parentState: hydroState)),
      ];
    });
    table['getProperties'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        maybeBoxObject<List<dynamic>>(
            object: vmObject
                .getProperties()
                .map((x) => maybeBoxObject<DiagnosticsNode>(
                    object: x, hydroState: hydroState, table: HydroTable()))
                .toList(),
            hydroState: hydroState,
            table: HydroTable()),
      ];
    });
    table['getChildren'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        maybeBoxObject<List<dynamic>>(
            object: vmObject
                .getChildren()
                .map((x) => maybeBoxObject<DiagnosticsNode>(
                    object: x, hydroState: hydroState, table: HydroTable()))
                .toList(),
            hydroState: hydroState,
            table: HydroTable()),
      ];
    });
    table['getPropertyType'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        maybeBoxObject<Type>(
            object: vmObject.propertyType,
            hydroState: hydroState,
            table: HydroTable()),
      ];
    });
    table['getValue'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      final returnValue = vmObject.value;
      if (returnValue != null) {
        return [
          maybeBoxObject<Object?>(
              object: returnValue, hydroState: hydroState, table: HydroTable()),
        ];
      }
      return [];
    });
    table['getException'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      final returnValue = vmObject.exception;
      if (returnValue != null) {
        return [
          maybeBoxObject<Object?>(
              object: returnValue, hydroState: hydroState, table: HydroTable()),
        ];
      }
      return [];
    });
    table['getIsInteresting'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        vmObject.isInteresting,
      ];
    });
    table['getLevel'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        DiagnosticLevel.values.indexWhere((x) {
          return x == vmObject.level;
        }),
      ];
    });
    table['isFiltered'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        vmObject.isFiltered(maybeUnBoxEnum(
            values: DiagnosticLevel.values, boxedEnum: luaCallerArguments[1])),
      ];
    });
    table['toString'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        vmObject.toString(
            minLevel: maybeUnBoxEnum(
                values: DiagnosticLevel.values,
                boxedEnum: luaCallerArguments.length >= 2
                    ? luaCallerArguments[1]['minLevel']
                    : null),
            parentConfiguration:
                maybeUnBoxAndBuildArgument<TextTreeConfiguration?, dynamic>(
                    luaCallerArguments.length >= 2
                        ? luaCallerArguments[1]['parentConfiguration']
                        : null,
                    parentState: hydroState)),
      ];
    });
    table['toStringDeep'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        vmObject.toStringDeep(
            minLevel: maybeUnBoxEnum(
                values: DiagnosticLevel.values,
                boxedEnum: luaCallerArguments.length >= 2
                    ? luaCallerArguments[1]['minLevel']
                    : null),
            parentConfiguration:
                maybeUnBoxAndBuildArgument<TextTreeConfiguration?, dynamic>(
                    luaCallerArguments.length >= 2
                        ? luaCallerArguments[1]['parentConfiguration']
                        : null,
                    parentState: hydroState),
            prefixLineOne: luaCallerArguments.length >= 2
                ? luaCallerArguments[1]['prefixLineOne']
                : null,
            prefixOtherLines: luaCallerArguments.length >= 2
                ? luaCallerArguments[1]['prefixOtherLines']
                : null),
      ];
    });
    table['getEmptyBodyDescription'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      final returnValue = vmObject.emptyBodyDescription;
      if (returnValue != null) {
        return [
          returnValue,
        ];
      }
      return [];
    });
    table['getAllowWrap'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        vmObject.allowWrap,
      ];
    });
    table['getAllowNameWrap'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        vmObject.allowNameWrap,
      ];
    });
    table['getAllowTruncate'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        vmObject.allowTruncate,
      ];
    });
    table['getHashCode'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        vmObject.hashCode,
      ];
    });
  }

  final HydroTable table;

  final HydroState hydroState;

  final DiagnosticsDebugCreator vmObject;
}

class RTManagedDiagnosticsDebugCreator extends DiagnosticsDebugCreator
    implements Box<DiagnosticsDebugCreator> {
  RTManagedDiagnosticsDebugCreator(Object value$,
      {required this.table, required this.hydroState})
      : super(
          value$,
        ) {
    table['vmObject'] = vmObject;
    table['unwrap'] = makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [unwrap()];
    });
    table['expandableValue'] = expandableValue;
    table['allowWrap'] = allowWrap;
    table['allowNameWrap'] = allowNameWrap;
    table['ifNull'] = ifNull;
    table['ifEmpty'] = ifEmpty;
    table['tooltip'] = tooltip;
    table['missingIfNull'] = missingIfNull;
    table['defaultValue'] = maybeBoxObject<Object?>(
        object: defaultValue, hydroState: hydroState, table: HydroTable());
    table['name'] = name;
    table['showSeparator'] = showSeparator;
    table['showName'] = showName;
    table['linePrefix'] = linePrefix;
    table['style'] = DiagnosticsTreeStyle.values.indexWhere((x) {
      return x == style;
    });
    table['_dart_toJsonMap'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        maybeBoxObject<Map>(
            object: super.toJsonMap(maybeUnBoxAndBuildArgument<
                DiagnosticsSerializationDelegate,
                dynamic>(luaCallerArguments[1], parentState: hydroState)),
            hydroState: hydroState,
            table: HydroTable())
      ];
    });
    table['_dart_valueToString'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        super.valueToString(
            parentConfiguration:
                maybeUnBoxAndBuildArgument<TextTreeConfiguration?, dynamic>(
                    luaCallerArguments.length >= 2
                        ? luaCallerArguments[1]['parentConfiguration']
                        : null,
                    parentState: hydroState))
      ];
    });
    table['_dart_toDescription'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        super.toDescription(
            parentConfiguration:
                maybeUnBoxAndBuildArgument<TextTreeConfiguration?, dynamic>(
                    luaCallerArguments.length >= 2
                        ? luaCallerArguments[1]['parentConfiguration']
                        : null,
                    parentState: hydroState))
      ];
    });
    table['_dart_getProperties'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        maybeBoxObject<List<dynamic>>(
            object: super
                .getProperties()
                .map((x) => maybeBoxObject<DiagnosticsNode>(
                    object: x, hydroState: hydroState, table: HydroTable()))
                .toList(),
            hydroState: hydroState,
            table: HydroTable())
      ];
    });
    table['_dart_getChildren'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        maybeBoxObject<List<dynamic>>(
            object: super
                .getChildren()
                .map((x) => maybeBoxObject<DiagnosticsNode>(
                    object: x, hydroState: hydroState, table: HydroTable()))
                .toList(),
            hydroState: hydroState,
            table: HydroTable())
      ];
    });
    table['_dart_getPropertyType'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [super.propertyType];
    });
    table['_dart_getValue'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [super.value];
    });
    table['_dart_getException'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [super.exception];
    });
    table['_dart_getIsInteresting'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [super.isInteresting];
    });
    table['_dart_getLevel'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [super.level];
    });
    table['_dart_isFiltered'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        super.isFiltered(maybeUnBoxEnum(
            values: DiagnosticLevel.values, boxedEnum: luaCallerArguments[1]))
      ];
    });
    table['_dart_toString'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        super.toString(
            minLevel: maybeUnBoxEnum(
                values: DiagnosticLevel.values,
                boxedEnum: luaCallerArguments.length >= 2
                    ? luaCallerArguments[1]['minLevel']
                    : null),
            parentConfiguration:
                maybeUnBoxAndBuildArgument<TextTreeConfiguration?, dynamic>(
                    luaCallerArguments.length >= 2
                        ? luaCallerArguments[1]['parentConfiguration']
                        : null,
                    parentState: hydroState))
      ];
    });
    table['_dart_toStringDeep'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        super.toStringDeep(
            minLevel: maybeUnBoxEnum(
                values: DiagnosticLevel.values,
                boxedEnum: luaCallerArguments.length >= 2
                    ? luaCallerArguments[1]['minLevel']
                    : null),
            parentConfiguration:
                maybeUnBoxAndBuildArgument<TextTreeConfiguration?, dynamic>(
                    luaCallerArguments.length >= 2
                        ? luaCallerArguments[1]['parentConfiguration']
                        : null,
                    parentState: hydroState),
            prefixLineOne: luaCallerArguments.length >= 2
                ? luaCallerArguments[1]['prefixLineOne']
                : null,
            prefixOtherLines: luaCallerArguments.length >= 2
                ? luaCallerArguments[1]['prefixOtherLines']
                : null)
      ];
    });
    table['_dart_getEmptyBodyDescription'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [super.emptyBodyDescription];
    });
    table['_dart_getAllowWrap'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [super.allowWrap];
    });
    table['_dart_getAllowNameWrap'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [super.allowNameWrap];
    });
    table['_dart_getAllowTruncate'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [super.allowTruncate];
    });
    table['_dart_getTextTreeConfiguration'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [super.textTreeConfiguration];
    });
    table['_dart_getHashCode'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [super.hashCode];
    });
  }

  final HydroTable table;

  final HydroState hydroState;

  DiagnosticsDebugCreator unwrap() => this;
  DiagnosticsDebugCreator get vmObject => this;
  @override
  Map<String, Object?> toJsonMap(delegate) {
    super.toJsonMap(delegate);
    Closure closure = table["toJsonMap"];
    return maybeUnBoxAndBuildArgument<Map<String, Object?>, String>(
        closure.dispatch([table], parentState: hydroState)[0],
        parentState: hydroState);
  }

  @override
  String valueToString({TextTreeConfiguration? parentConfiguration}) {
    Closure closure = table["valueToString"];
    return closure.dispatch([table], parentState: hydroState)[0];
  }

  @override
  String toDescription({TextTreeConfiguration? parentConfiguration}) {
    Closure closure = table["toDescription"];
    return closure.dispatch([table], parentState: hydroState)[0];
  }

  @override
  List<DiagnosticsNode> getProperties() {
    Closure closure = table["getProperties"];
    return maybeUnBoxAndBuildArgument<List<DiagnosticsNode>, DiagnosticsNode>(
        closure.dispatch([table], parentState: hydroState)[0],
        parentState: hydroState);
  }

  @override
  List<DiagnosticsNode> getChildren() {
    Closure closure = table["getChildren"];
    return maybeUnBoxAndBuildArgument<List<DiagnosticsNode>, DiagnosticsNode>(
        closure.dispatch([table], parentState: hydroState)[0],
        parentState: hydroState);
  }

  @override
  Type get propertyType {
    Closure closure = table["getPropertyType"];
    return maybeUnBoxAndBuildArgument<Type, dynamic>(
        closure.dispatch([table], parentState: hydroState)[0],
        parentState: hydroState);
  }

  @override
  Object? get value {
    Closure closure = table["getValue"];
    return maybeUnBoxAndBuildArgument<Object?, dynamic>(
        closure.dispatch([table], parentState: hydroState)[0],
        parentState: hydroState);
  }

  @override
  Object? get exception {
    Closure closure = table["getException"];
    return maybeUnBoxAndBuildArgument<Object?, dynamic>(
        closure.dispatch([table], parentState: hydroState)[0],
        parentState: hydroState);
  }

  @override
  bool get isInteresting {
    Closure closure = table["getIsInteresting"];
    return closure.dispatch([table], parentState: hydroState)[0];
  }

  @override
  DiagnosticLevel get level {
    Closure closure = table["getLevel"];
    return maybeUnBoxEnum(
        values: DiagnosticLevel.values,
        boxedEnum: closure.dispatch([table], parentState: hydroState)[0]);
  }

  @override
  bool isFiltered(minLevel) {
    Closure closure = table["isFiltered"];
    return closure.dispatch([table], parentState: hydroState)[0];
  }

  @override
  String toString(
      {DiagnosticLevel minLevel = DiagnosticLevel.info,
      TextTreeConfiguration? parentConfiguration}) {
    Closure closure = table["__tostring"];
    return closure.dispatch([table], parentState: hydroState)[0];
  }

  @override
  String toStringDeep(
      {DiagnosticLevel minLevel = DiagnosticLevel.debug,
      TextTreeConfiguration? parentConfiguration,
      String prefixLineOne = '',
      String? prefixOtherLines}) {
    Closure closure = table["toStringDeep"];
    return closure.dispatch([table], parentState: hydroState)[0];
  }

  @override
  String? get emptyBodyDescription {
    Closure closure = table["getEmptyBodyDescription"];
    return closure.dispatch([table], parentState: hydroState)[0];
  }

  @override
  bool get allowWrap {
    Closure closure = table["getAllowWrap"];
    return closure.dispatch([table], parentState: hydroState)[0];
  }

  @override
  bool get allowNameWrap {
    Closure closure = table["getAllowNameWrap"];
    return closure.dispatch([table], parentState: hydroState)[0];
  }

  @override
  bool get allowTruncate {
    Closure closure = table["getAllowTruncate"];
    return closure.dispatch([table], parentState: hydroState)[0];
  }

  @override
  TextTreeConfiguration? get textTreeConfiguration {
    Closure closure = table["getTextTreeConfiguration"];
    return maybeUnBoxAndBuildArgument<TextTreeConfiguration?, dynamic>(
        closure.dispatch([table], parentState: hydroState)[0],
        parentState: hydroState);
  }

  @override
  int get hashCode {
    Closure closure = table["getHashCode"];
    return closure.dispatch([table], parentState: hydroState)[0];
  }
}

void loadDiagnosticsDebugCreator(
    {required HydroState hydroState, required HydroTable table}) {
  table['diagnosticsDebugCreator'] =
      makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
    return [
      RTManagedDiagnosticsDebugCreator(
          maybeUnBoxAndBuildArgument<Object, dynamic>(luaCallerArguments[1],
              parentState: hydroState),
          table: luaCallerArguments[0],
          hydroState: hydroState)
    ];
  });
  registerBoxer<DiagnosticsDebugCreator>(boxer: (
      {required DiagnosticsDebugCreator vmObject,
      required HydroState hydroState,
      required HydroTable table}) {
    return VMManagedDiagnosticsDebugCreator(
        vmObject: vmObject, hydroState: hydroState, table: table);
  });
}
